fix(jans-auth-server): ClassCastException during select account

**Describe the bug**

Got ClassCastException during select account.
```
Caused by: java.lang.ClassCastException: class org.eclipse.jetty.server.Request cannot be cast to class jakarta.servlet.http.HttpServletResponse (org.eclipse.jetty.server.Request and jakarta.servlet.http.HttpServletResponse are in unnamed module of loader org.eclipse.jetty.start.Classpath$Loader @36b4cef0)
        at io.jans.as.server.model.common.ExecutionContext.of(ExecutionContext.java:104) ~[classes/:?]
        at io.jans.as.server.auth.SelectAccountAction.prepare(SelectAccountAction.java:130) ~[classes/:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.jboss.weld.injection.producer.DefaultLifecycleCallbackInvoker.invokeMethods(DefaultLifecycleCallbackInvoker.java:83) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.injection.producer.DefaultLifecycleCallbackInvoker.postConstruct(DefaultLifecycleCallbackInvoker.java:66) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.injection.producer.BasicInjectionTarget.postConstruct(BasicInjectionTarget.java:122) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:164) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.contexts.AbstractContext.get(AbstractContext.java:96) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.bean.ContextualInstanceStrategy$CachingContextualInstanceStrategy.get(ContextualInstanceStrategy.java:177) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:694) ~[weld-core-impl-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.module.web.el.AbstractWeldELResolver.lookup(AbstractWeldELResolver.java:107) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.module.web.el.AbstractWeldELResolver.getValue(AbstractWeldELResolver.java:90) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
        at org.jboss.weld.environment.servlet.util.ForwardingELResolver.getValue(ForwardingELResolver.java:49) ~[weld-servlet-core-4.0.3.Final.jar:4.0.3.Final]
        at jakarta.el.CompositeELResolver.getValue(CompositeELResolver.java:62) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
        at com.sun.faces.el.DemuxCompositeELResolver._getValue(DemuxCompositeELResolver.java:139) ~[jakarta.faces-3.0.2.jar:3.0.2]
        at com.sun.faces.el.DemuxCompositeELResolver.getValue(DemuxCompositeELResolver.java:164) ~[jakarta.faces-3.0.2.jar:3.0.2]
        at org.apache.el.parser.AstIdentifier.getValue(AstIdentifier.java:93) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
        at org.apache.el.parser.AstValue.getTarget(AstValue.java:69) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
        at org.apache.el.parser.AstValue.getType(AstValue.java:57) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
        at org.apache.el.ValueExpressionImpl.getType(ValueExpressionImpl.java:173) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
        at org.jboss.weld.module.web.el.WeldValueExpression.getType(WeldValueExpression.java:93) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
        at com.sun.faces.facelets.el.TagValueExpression.getType(TagValueExpression.java:62) ~[jakarta.faces-3.0.2.jar:3.0.2]
        at com.sun.faces.renderkit.html_basic.HtmlBasicInputRenderer.getConvertedValue(HtmlBasicInputRenderer.java:69) ~[jakarta.faces-3.0.2.jar:3.0.2]
        at jakarta.faces.component.UIViewParameter.getConvertedValue(UIViewParameter.java:431) ~[jakarta.faces-3.0.2.jar:3.0.2]

```

--- a/jans-auth-server/client/src/test/java/io/jans/as/client/ciba/CibaPingModeJwtAuthRequestTests.java
+++ b/jans-auth-server/client/src/test/java/io/jans/as/client/ciba/CibaPingModeJwtAuthRequestTests.java
@@ -340,6 +340,7 @@ public class CibaPingModeJwtAuthRequestTests extends BaseTest {
         showTitle("idTokenHintRS384");
 
         List<ResponseType> responseTypes = Arrays.asList(ResponseType.TOKEN, ResponseType.ID_TOKEN);
+        List<String> scopes = Collections.singletonList("openid");
 
         // 1. Register client
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
@@ -347,6 +348,7 @@ public class CibaPingModeJwtAuthRequestTests extends BaseTest {
         registerRequest.setResponseTypes(responseTypes);
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
         registerRequest.setIdTokenSignedResponseAlg(SignatureAlgorithm.RS384);
+        registerRequest.setScope(scopes);
 
         RegisterClient registerClient = new RegisterClient(registrationEndpoint);
         registerClient.setRequest(registerRequest);
@@ -358,7 +360,6 @@ public class CibaPingModeJwtAuthRequestTests extends BaseTest {
         String clientId = registerResponse.getClientId();
 
         // 2. Request authorization
-        List<String> scopes = Collections.singletonList("openid");
         String nonce = UUID.randomUUID().toString();
         String state = UUID.randomUUID().toString();
 

--- a/jans-auth-server/client/src/test/java/io/jans/as/client/ciba/CibaPollModeJwtAuthRequestTests.java
+++ b/jans-auth-server/client/src/test/java/io/jans/as/client/ciba/CibaPollModeJwtAuthRequestTests.java
@@ -313,6 +313,7 @@ public class CibaPollModeJwtAuthRequestTests extends BaseTest {
         showTitle("idTokenHintRS384");
 
         List<ResponseType> responseTypes = Arrays.asList(ResponseType.TOKEN, ResponseType.ID_TOKEN);
+        List<String> scopes = Collections.singletonList("openid");
 
         // 1. Register client
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
@@ -320,6 +321,7 @@ public class CibaPollModeJwtAuthRequestTests extends BaseTest {
         registerRequest.setResponseTypes(responseTypes);
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
         registerRequest.setIdTokenSignedResponseAlg(SignatureAlgorithm.RS384);
+        registerRequest.setScope(scopes);
 
         RegisterClient registerClient = new RegisterClient(registrationEndpoint);
         registerClient.setRequest(registerRequest);
@@ -331,7 +333,6 @@ public class CibaPollModeJwtAuthRequestTests extends BaseTest {
         String clientId = registerResponse.getClientId();
 
         // 2. Request authorization
-        List<String> scopes = Collections.singletonList("openid");
         String nonce = UUID.randomUUID().toString();
         String state = UUID.randomUUID().toString();
 

--- a/jans-auth-server/client/src/test/java/io/jans/as/client/interop/CanProvideEncryptedUserInfoResponse.java
+++ b/jans-auth-server/client/src/test/java/io/jans/as/client/interop/CanProvideEncryptedUserInfoResponse.java
@@ -42,6 +42,7 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         showTitle("OC5:FeatureTest-Can Provide Encrypted UserInfo Response A128KW A128GCM");
 
         List<ResponseType> responseTypes = Arrays.asList(ResponseType.TOKEN, ResponseType.ID_TOKEN);
+        List<String> scopes = Arrays.asList("openid", "profile", "address", "email");
 
         // 1. Register client
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
@@ -50,6 +51,7 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         registerRequest.setUserInfoEncryptedResponseAlg(KeyEncryptionAlgorithm.A128KW);
         registerRequest.setUserInfoEncryptedResponseEnc(BlockEncryptionAlgorithm.A128GCM);
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
+        registerRequest.setScope(scopes);
 
         RegisterClient registerClient = new RegisterClient(registrationEndpoint);
         registerClient.setRequest(registerRequest);
@@ -62,7 +64,6 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         String clientSecret = registerResponse.getClientSecret();
 
         // 2. Request authorization
-        List<String> scopes = Arrays.asList("openid", "profile", "address", "email");
         String nonce = UUID.randomUUID().toString();
         String state = UUID.randomUUID().toString();
 
@@ -101,6 +102,7 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         showTitle("OC5:FeatureTest-Can Provide Encrypted UserInfo Response A256KW A256GCM");
 
         List<ResponseType> responseTypes = Arrays.asList(ResponseType.TOKEN, ResponseType.ID_TOKEN);
+        List<String> scopes = Arrays.asList("openid", "profile", "address", "email");
 
         // 1. Register client
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
@@ -109,6 +111,7 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         registerRequest.setUserInfoEncryptedResponseAlg(KeyEncryptionAlgorithm.A256KW);
         registerRequest.setUserInfoEncryptedResponseEnc(BlockEncryptionAlgorithm.A256GCM);
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
+        registerRequest.setScope(scopes);
 
         RegisterClient registerClient = new RegisterClient(registrationEndpoint);
         registerClient.setRequest(registerRequest);
@@ -121,7 +124,6 @@ public class CanProvideEncryptedUserInfoResponse extends BaseTest {
         String clientSecret = registerResponse.getClientSecret();
 
         // 2. Request authorization
-        List<String> scopes = Arrays.asList("openid", "profile", "address", "email");
         String nonce = UUID.randomUUID().toString();
         String state = UUID.randomUUID().toString();
 

--- a/jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/ResponseTypesRestrictionHttpTest.java
+++ b/jans-auth-server/client/src/test/java/io/jans/as/client/ws/rs/ResponseTypesRestrictionHttpTest.java
@@ -445,8 +445,8 @@ public class ResponseTypesRestrictionHttpTest extends BaseTest {
                 .check();
     }
 
-    @DataProvider(name = "responseTypesTokenIdTokenFailDataProvider")
-    public Object[][] responseTypesTokenIdTokenFailDataProvider(ITestContext context) {
+    @DataProvider(name = "responseTypesTokenIdTokenDataProvider")
+    public Object[][] responseTypesTokenIdTokenDataProvider(ITestContext context) {
         String redirectUris = context.getCurrentXmlTest().getParameter("redirectUris");
         String redirectUri = context.getCurrentXmlTest().getParameter("redirectUri");
         String userId = context.getCurrentXmlTest().getParameter("userId");
@@ -461,8 +461,8 @@ public class ResponseTypesRestrictionHttpTest extends BaseTest {
         };
     }
 
-    @Test(dataProvider = "responseTypesTokenIdTokenFailDataProvider")
-    public void responseTypesTokenIdTokenFail(
+    @Test(dataProvider = "responseTypesTokenIdTokenDataProvider")
+    public void responseTypesTokenIdToken(
             final String redirectUris, final String redirectUri, final String userId, final String userSecret,
             final List<ResponseType> responseTypes, final String sectorIdentifierUri) throws Exception {
         showTitle("responseTypesTokenIdTokenFail");
@@ -477,7 +477,6 @@ public class ResponseTypesRestrictionHttpTest extends BaseTest {
         RegisterRequest registerRequest = new RegisterRequest(ApplicationType.WEB, "jans test app",
                 StringUtils.spaceSeparatedToList(redirectUris));
         registerRequest.setResponseTypes(Arrays.asList(
-                ResponseType.TOKEN,
                 ResponseType.ID_TOKEN));
         registerRequest.setSectorIdentifierUri(sectorIdentifierUri);
         registerRequest.setScope(scopes);
@@ -488,6 +487,7 @@ public class ResponseTypesRestrictionHttpTest extends BaseTest {
 
         showClient(registerClient);
         AssertBuilder.registerResponse(registerResponse).created().check();
+        assertTrue(registerResponse.getResponseTypes().contains(ResponseType.CODE));
 
         String clientId = registerResponse.getClientId();
         String registrationAccessToken = registerResponse.getRegistrationAccessToken();
@@ -523,7 +523,5 @@ public class ResponseTypesRestrictionHttpTest extends BaseTest {
         showClient(authorizeClient);
         assertTrue(authorizationResponse.getStatus() == 302
                 || authorizationResponse.getStatus() == 400, "Unexpected response code: " + authorizationResponse.getStatus());
-        assertNotNull(authorizationResponse.getErrorType(), "The error type is null");
-        assertNotNull(authorizationResponse.getErrorDescription(), "The error description is null");
     }
 }

--- a/jans-auth-server/server/src/main/java/io/jans/as/server/model/common/ExecutionContext.java
+++ b/jans-auth-server/server/src/main/java/io/jans/as/server/model/common/ExecutionContext.java
@@ -101,7 +101,7 @@ public class ExecutionContext {
                 executionContext.setHttpRequest((HttpServletRequest) externalContext.getRequest());
             }
             if (externalContext.getResponse() instanceof HttpServletResponse) {
-                executionContext.setHttpResponse((HttpServletResponse) externalContext.getRequest());
+                executionContext.setHttpResponse((HttpServletResponse) externalContext.getResponse());
             }
         }
         return executionContext;

--- a/jans-linux-setup/jans_setup/templates/test/jans-auth/data/oxauth-test-data.ldif
+++ b/jans-linux-setup/jans_setup/templates/test/jans-auth/data/oxauth-test-data.ldif
@@ -97,6 +97,11 @@ changetype: modify
 jansDefScope: true
 replace: jansDefScope
 
+dn: inum=B9D2-D6E5,ou=scopes,o=jans
+changetype: modify
+jansDefScope: true
+replace: jansDefScope
+
 dn: inum=341A,ou=scopes,o=jans
 changetype: modify
 add: jansClaim

